### PR TITLE
Feature/py3

### PR DESCRIPTION
**Pull request by jbrown-xentity**
Related to https://github.com/GSA/datagov-deploy/issues/3401

Python3 image being utilized and system is responding. Tests passing, and successful deployment to development:
https://github.com/GSA/catalog.data.gov/runs/3509039946?check_suite_focus=true

### CODE REVIEW

File: ckan/freeze-requirements.sh [hunk 1]
```diff
@@ -11,17 +11,22 @@ set -e
...
-# sadness -- pyz3950 is so old that poetry can't make sense of it's setup.py, so
-# we have to add the requirement manually
-
-echo "-e git+https://github.com/asl2/PyZ3950.git@c2282c73182cef2beca0f65b1eb7699c9b24512e#egg=PyZ3950" >> requirements.txt
+# Sadness: poetry does not keep the "editable" mode that datagovcatalog is installed in.
+# Replace the line with the necessary editable flag
+sed -i 's/ckanext-datagovcatalog \@ git+https\:\/\/github.com\/GSA\/ckanext-datagovcatalog.git\@main/-e git+https:\/\/github.com\/GSA\/ckanext-datagovcatalog.git@main#egg=ckanext-datagovcatalog/g' requirements.txt
+# TODO: add the rest of the editably installed extensions
+# TODO: poetry does not pin to a github commit if given a branch;
+#       need to figure out how to pin appropriately
```
Comment (nickumia-reisys) [inline ckan/freeze-requirements.sh:29]: @jbrown-xentity Why would pinning to a branch translate to pinning to a commit.  If the branch is updated, isn't the normal functionality to use the most recent commit on that branch?

Comment (jbrown-xentity) [reply ckan/freeze-requirements.sh:29]: What we don't want to happen is for something to be able to build and/or deploy without being tested. Pinning to a commit solves that problem. We already have major changes that need to be made to datajson and datagovtheme that were not caught in their own tests...

**Comment by jbrown-xentity** [comment]
We'll have to change the required checks, but I believe this is ready for deployment...

**Comment by jbrown-xentity** [comment]
This is ready for final review. @mogul , we will need to change the required tests (I changed the name in GitHub actions to be more accurate). `Build and Test` and `vendor dependencies` should both be required before merging to main.

**Comment by jbrown-xentity** [comment]
Deploy working in develop is [here](https://github.com/GSA/catalog.data.gov/actions/runs/1222233335)

**Comment by jbrown-xentity** [comment]
Final test (pending) is here: https://github.com/GSA/catalog.data.gov/actions/runs/1222420404

**Review by nickumia-reisys — APPROVED** [review]
This was a lot of work, but all of the py3 stuff (requirements, cypress, local dev changes) looks good.  @mogul You might want to just look over the cf stuff 😄

**Review by mogul — APPROVED** [review]

**Pull request merged**